src/genie/libs/parser/iosxe/show_gnxi.py

```diff
@@ -0,0 +1,243 @@
+                "oper_state": {
+                    "grpc": {},
+                    "config_svc": {},
+                    "telemetry_svc": {},
+                    "cert_mgmt_svc": {},
+                    "os_image_svc": {},
+                    "factory_reset_svc": {}
+                    }
+                }
+
+        p_full = re.compile(r"Settings\n"
+                r"=*\n"
+                r"\s\sServer: (?P<insecure_server>Disabled|Enabled)\n"
+                r"\s\sServer port: (?P<insecure_port>\d+)\n"
+                r"\s\sSecure server: (?P<secure_server>Disabled|Enabled)\n"
+                r"\s\sSecure server port: (?P<secure_port>\d+)\n"
+                r"\s\sSecure client authentication: (?P<secure_client_auth>Disabled|Enabled)\n"
+                r"\s\sSecure trustpoint:\s?(?P<secure_trustpoint>.*)\n"
+                r"\s\sSecure client trustpoint:\s?(?P<secure_client_trustpoint>.*)\n"
+                r"\s\sSecure password authentication: (?P<secure_pass_auth>Disabled|Enabled)\n\n"
+                r"GNMI\n"
+                r"====\n"
+                r"\s\sAdmin state: (?P<admin_state>Disabled|Enabled)\n"
+                r"\s\sOper status: (?P<oper_status>Down|Up)\n"
+                r"\s\sState: (?P<bootstrapping_state>Provisioned|Default)\n\n"
+                r"\s\sgRPC Server\n"
+                r"\s\s-*\n"
+                r"\s\s\s\sAdmin state: (?P<grpc_admin_state>Disabled|Enabled)\n"
+                r"\s\s\s\sOper status: (?P<grpc_oper_status>Down|Up)\n\n"
+                r"\s\sConfiguration service\n"
+                r"\s\s-*\n"
+                r"\s\s\s\sAdmin state: (?P<config_svc_admin_state>Disabled|Enabled)\n"
+                r"\s\s\s\sOper status: (?P<config_svc_oper_status>Down|Up)\n\n"
+                r"\s\sTelemetry service\n"
+                r"\s\s-*\n"
+                r"\s\s\s\sAdmin state: (?P<telemetry_svc_admin_state>Disabled|Enabled)\n"
+                r"\s\s\s\sOper status: (?P<telemetry_svc_oper_status>Down|Up)\n\n"
+                r"GNOI\n"
+                r"====\n\n"
+                r"\s\sCert Management service\n"
+                r"\s\s-*\n"
+                r"\s\s\s\sAdmin state: (?P<cert_svc_admin_state>Disabled|Enabled)\n"
+                r"\s\s\s\sOper status: (?P<cert_svc_oper_status>Down|Up)\n\n"
+                r"\s\sOS Image service\n"
+                r"\s\s-*\n"
+                r"\s\s\s\sAdmin state: (?P<os_svc_admin_state>Disabled|Enabled)\n"
+                r"\s\s\s\sOper status: (?P<os_svc_oper_status>Down|Up)\n"
+                r"\s\s\s\sSupported: (?P<os_svc_supported>Not supported on this platform|Supported)\n\n"
+                r"\s\sFactory Reset service\n"
+                r"\s\s-*\n"
+                r"\s\s\s\sAdmin state: (?P<reset_svc_admin_state>Disabled|Enabled)\n"
+                r"\s\s\s\sOper status: (?P<reset_svc_oper_status>Down|Up)\n"
+                r"\s\s\s\sSupported: (?P<reset_svc_supported>Not supported on this platform|Supported)")
```

Comment:
We can't accept this. We parse our CLI outputs line by line. This is necessary because CLI output can change as new versions of the OS are released. One master regex is far too difficult to debug and maintain. Please look at other parsers for examples. The same goes for `ShowGnxiState` as well.

You need to put each regex in its own variable then match them line by line in a loop

Reply:
`ShowGnxiState` is a one liner already.

Reply:
it means each line of output should have its own regex and be matched.

Reply:
Please review other parsers to see what we expect

https://github.com/CiscoTestAutomation/genieparser/blob/73bb15dc83ed90bc38c683e33098e987b843509d/src/genie/libs/parser/iosxe/show_eigrp.py#L126
https://github.com/CiscoTestAutomation/genieparser/blob/73bb15dc83ed90bc38c683e33098e987b843509d/src/genie/libs/parser/iosxe/show_ip.py#L73